sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -983,6 +983,32 @@ object WindowSpecCheck extends ExprChecks {
   }
 }
 
+object CreateMapCheck extends ExprChecks {
+
+  // Spark supports all types except for Map for key and value (Map is not supported
+  // even in nested types)
+  val keyValueSig: TypeSig = (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64 +
```

Comment:
So we are not going to support a map as a value in a map? I understand the key, but why also the value?

Reply:
Thanks. I had misread the Spark documentation around this. I will add separate signatures for key and value.